### ISSUE TITLE
make is failing with missing kernel

### ISSUE THREAD

**Issue by kamsad** [sonyxperiadev/local_manifests#142]
In Debian bookworm, trying to build kernel for Xperia 5 IV but failing:

```
$ make -j2
01:24:22 ************************************************************
01:24:22 You are building on a machine with 15.3GB of RAM
01:24:22 
01:24:22 The minimum required amount of free memory is around 16GB,
01:24:22 and even with that, some configurations may not work.
01:24:22 
01:24:22 If you run into segfaults or other errors, try reducing your
01:24:22 -j value.
01:24:22 ************************************************************
build/make/core/soong_config.mk:209: warning: BOARD_PLAT_PUBLIC_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PUBLIC_SEPOLICY_DIRS instead.
build/make/core/soong_config.mk:210: warning: BOARD_PLAT_PRIVATE_SEPOLICY_DIR has been deprecated. Use SYSTEM_EXT_PRIVATE_SEPOLICY_DIRS instead.
============================================
PLATFORM_VERSION_CODENAME=REL
PLATFORM_VERSION=13
TARGET_PRODUCT=aosp_xqcq54
TARGET_BUILD_VARIANT=eng
TARGET_BUILD_TYPE=release
TARGET_ARCH=arm64
TARGET_ARCH_VARIANT=armv8-2a
TARGET_CPU_VARIANT=generic
TARGET_2ND_ARCH=arm
TARGET_2ND_ARCH_VARIANT=armv8-2a
TARGET_2ND_CPU_VARIANT=generic
HOST_ARCH=x86_64
HOST_2ND_ARCH=x86
HOST_OS=linux
HOST_OS_EXTRA=Linux-6.1.0-9-amd64-x86_64-Debian-GNU/Linux-12-(bookworm)
HOST_CROSS_OS=windows
HOST_CROSS_ARCH=x86
HOST_CROSS_2ND_ARCH=x86_64
HOST_BUILD_TYPE=release
BUILD_ID=TQ2A.230505.002
OUT_DIR=out
PRODUCT_SOONG_NAMESPACES=device/sony/common device/sony/nagara vendor/qcom/opensource/core-utils vendor/qcom/opensource/display/sm8450 vendor/qcom/opensource/display-commonsys-intf/sm8450
============================================
out/target/product/pdx224/clean_steps.mk was modified, regenerating...
FAILED: ninja: 'kernel/sony/msm-5.10/common-kernel/kernel', needed by 'out/target/product/pdx224/kernel', missing and no known rule to make it
01:24:44 ninja failed with: exit status 1

#### failed to build some targets (22 seconds) ####

```

**Comment by jerpelea** [comment]
cd kernel/sony/msm-5.10/common-kernel 
./build-kernels-clang.sh

**Comment by MarijnS95** [comment]
@kamsad please note that the prebuilt kernel is *intentionally not available* because it is a work in progress **AND CURRENTLY CAUSES MASSIVE OVERHEATING**. Be careful to not run the device for too long and burn yourself, or worse.

**Comment by kamsad** [comment]
Thank you guys. When should we expect the 'overheating' issue to be fixed and what can we do in the meanwhile? Avoid 13.0? Any recommendation would help.

PS: I did as told (had to do a couple of extra steps) and 'make' is working as I write this.

**Comment by MarijnS95** [comment]
"When" is not really a thing that exists in open-source communities, unfortunately. It'll get done when someone with the right skillset finds time to look into and solve the issue.

> Avoid 13.0?

Android 13 is the first revision where this platform gained support, so it's not possible to go back in time. Even then this is likely a kernel, HAL or device configuration issue, and those tend to carry over through the Android versions.

**Comment by kamsad** [comment]
So heating issue does not affect factory shipped 13.0 - only AOSP devices?  Otherwise, why wouldn't Sony spend time/resources in fixing this issue?

**Comment by kamsad** [comment]
Another question; do I have to flash all the partitions listed [here](https://developer.sony.com/develop/open-devices/guides/aosp-build-instructions/build-aosp-android-13#tutorial-step-4) for my Xperia 5 IV? Namely;

fastboot flash boot out/target/product/<device>/boot.img​
fastboot flash vbmeta out/target/product/<device>/vbmeta.img​
fastboot flash dtbo out/target/product/<device>/dtbo.img​
fastboot flash recovery out/target/product/<device>/recovery.img​
fastboot flash system out/target/product/<device>/system.img​
fastboot flash vendor out/target/product/<device>/vendor.img​
fastboot flash userdata out/target/product/<device>/userdata.img​
fastboot flash product out/target/product/<device>/product.img​​
fastboot flash vbmeta_system out/target/product/<device>/vbmeta_system.img
fastboot flash system_ext out/target/product/<device>/system_ext.img
fastboot flash vendor_boot out/target/product/<device>/vendor_boot.img

Curious to know what each of these do? How do I know which ones my device needs? How do get a list of partitions my device currently has (Just flashed it with 'Emma' and it only had Android 12 image). Any link to a document or explanation would help. Thank you!
